### ISSUE TITLE
Create cross platform symlinks

### ISSUE THREAD

**Issue by CohenArthur** [Skallwar/suckit#49]

**Comment by SpyrosRoum** [comment]
destination will always be a directory, right?

**Comment by Skallwar** [comment]
> destination will always be a directory, right?

No, a file

**Comment by SpyrosRoum** [comment]
Hm to be honest I am not really sure I get why you need to create symlinks at all but assuming it's always for a file, I can do that with [windows::os::fs::symlink_file](https://doc.rust-lang.org/std/os/windows/fs/fn.symlink_file.html)

**Comment by Skallwar** [comment]
> Hm to be honest I am not really sure I get why you need to create symlinks at all but assuming it's always for a file, I can do that with [windows::os::fs::symlink_file](https://doc.rust-lang.org/std/os/windows/fs/fn.symlink_file.html)

Sometime we discover a link wich is a pdf (eg. test.org/pdf/test.pdf). So a page wich contain a link to this file will replace every test.org/pdf/test.pdf by test_org_pdf_test_pdf. But the real name of this pdf (test.pdf) is only found when actualy processing the url (test.org/pdf/test.pdf). At this point we save the pdf to it's real filename and create a symlink between test.pdf and ```test_org_pdf_test_pdf```

**Comment by SpyrosRoum** [comment]
If you know of a url that needs this I can try to do it. Is that the only reason why it doesn't work in windows?

**Comment by CohenArthur** [comment]
We *think* it's the only reason but haven't tried. There might be more to it. If you have a windows build available, feel free to try. I'll add it to the CI as well

**Comment by SpyrosRoum** [comment]
I have a vm I can use but from what I understand it doesn't happen with every page you try to download, right?

**Comment by CohenArthur** [comment]
True, symlinks aren't always created. The code calling `disk::symlink()` is [here](https://github.com/Skallwar/suckit/blob/16a5ed9cf73cc3446822e0b6c654219ea3c81abc/src/scraper.rs#L131)

**Comment by SpyrosRoum** [comment]
I know, I am asking if you already know of a page that requires symlinks so I can easily test it

**Comment by CohenArthur** [comment]
Oh my bad, sorry about the confusion. Any page downloading PDFs basically, so a page like https://dl.acm.org/doi/book/10.1145/3368274 is good.
